pkg/operators/oci-handler/oci.go

```diff
@@ -256,6 +263,37 @@ func (o *OciHandlerInstance) init(gadgetCtx operators.GadgetContext) error {
 		return fmt.Errorf("unmarshalling metadata: %w", err)
 	}
 
+	for _, ann := range o.ociParams.Get("annotate").AsStringSlice() {
+		if len(ann) == 0 {
+			continue
+		}
+		annInfo := strings.SplitN(ann, ":", 2)
+		if len(annInfo) != 2 {
+			return fmt.Errorf("invalid annotation %q", ann)
+		}
+
+		annotation := strings.SplitN(annInfo[1], "=", 2)
+		if len(annotation) != 2 {
+			return fmt.Errorf("invalid annotation %q", ann)
+		}
+
+		subject := strings.Split(annInfo[0], ".")
+		switch len(subject) {
+		case 1:
+			// data source
+			current := viper.GetStringMapString("datasources." + annInfo[0] + ".annotations")
+			current[annotation[0]] = annotation[1]
+			viper.Set("datasources."+subject[0]+".annotations", current)
+			log.Debugf("ds annotation %q added", ann)
+		case 2:
+			// field
+			current := viper.GetStringMapString("datasources." + subject[0] + ".fields." + subject[1] + ".annotations")
```

Comment:
What about fields with parent like "runtime.containername"?

Reply:
handled in #3468